### PR TITLE
socket_ncs: remove deprecated nrf_modem AT socket 

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
The AT socket has been removed from the modem library after being deprecated
for two subsequent releases. These AT socket related macros are therefore
no longer required.

Signed-off-by: Mirko Covizzi <mirko.covizzi@nordicsemi.no>

### CODE REVIEW

File: include/net/socket_ncs.h [hunk 1]
```diff
@@ -16,18 +16,6 @@
 extern "C" {
 #endif
 
-/* NCS specific protocol/address families. */
-
-#define PF_LTE 102 /**< Protocol family specific to LTE. */
-#define AF_LTE PF_LTE /**< Address family specific to LTE. */
```
Comment (eivindj-nordic) [inline include/net/socket_ncs.h]: There is one occurrence of `AF_LTE` and `NPROTO_AT` in `nrf/lib/at_cmd/at_cmd.c` that I don't see in the sdk-nrf PR [[PR-36]](https://github.com/lemrey/sdk-nrf/pull/36/files) Is anyone working on rewriting/removing this?

**Review by anhmolt — APPROVED** [review]

File: include/net/socket_ncs.h (hunk 1 above)
Comment (lemrey) [reply include/net/socket_ncs.h]: That library will be removed before we release `v1.5.0`

**Comment by lemrey** [comment]
@anhmolt could you add a commit here to remove the DFU socket defines in that file?

**Comment by anhmolt** [comment]
Didn't have push access, so I made a pull-request to [MirkoCovizzi:NCSDK-10724](https://github.com/MirkoCovizzi/sdk-zephyr/pull/1) that you can merge.

Wonder what happend to: https://github.com/lemrey/sdk-zephyr/pull/1 😕. A rebase I would guess.

**Comment by lemrey** [comment]
mm indeed, never mind then I can cherry pick that commit.

**Comment by lemrey** [comment]
I rebased the `libmodem` branch and picked this commit directly, so am closing this.

**Pull request closed without merging**